Dark theme

AshleyHenry15 opened the pull request:
This PR adds a dark theme and resolves other theme-related issues:

Resolves #57 
Resolves #7 

Known issues:
#56 
#59 
Flashing: https://github.com/quarto-dev/quarto-cli/issues/1325

Outstanding items tracking:
- [x] Nav bar drop-down menu: update hover color


Before we merge:
- [x] Update changelog
- [x] Clean up CSS (make sure that I don't have duplicates, that the style sheets follow the same order/org, etc.)
- [x] Maybe clean up examples in files? 

Link to view docs: 
https://ashley.quarto.pub/theme-testing/

edavidaja reviewed (approved):
more flash than a website in 1999

m-- commented on _extensions/posit-docs/theme-dark.scss:
```diff
@@ -0,0 +1,395 @@
+//font and body color
+
+$body-color: $dark-md-body;
+$popover-bg: $posit-green !default; // check this - citation previews
```
```
// check this - citation previews
```

does something here still need to be checked?

AshleyHenry15 replied:
Verified and comment removed! Thank you!

m-- commented on _extensions/posit-docs/theme-dark.scss:
```diff
@@ -0,0 +1,395 @@
+
+/* Code block elements */
+
+pre > code.sourceCode > span {
+  //color: #003B4F;
+}
+
+code span.ex {
+    //font-weight: bold;
+    color: #ffffff;
+}
+
+code span.dt {
+  color: #FFABCA;
+    //font-style: inherit;
+}
+
+code span.ot {
+    //color: rgb(60, 222, 222);
+}
+
+code span.st {
+    //color: greenyellow;
+}
```
for the commented-out bits here: do these need to stay (i.e. they are part of something coming soon), or can they be removed?

AshleyHenry15 replied:
I think that I had added the styles assuming that they would need updates but I think that I am safe to remove

m-- commented on _extensions/posit-docs/theme-dark.scss:
```diff
@@ -0,0 +1,395 @@
+/* Code block background */
+$code-block-bg: lighten($dark-md-background, 10%);
+
+//$monobackgroundcolor: $dark-md-background !default;
```
here too re: commented out code?

AshleyHenry15 replied:
I think that I am ok to remove this as well

m-- reviewed (approved):
It looks like there are some commented-out lines in the dark mode `.scss` file, not sure if those are still needed or not? Also, there's some inconsistent whitespace formatting in that file, not sure if you want to fix that (either as part of this, or at some point in the future), but i thought i'd mention it. Hooray dark mode! 😎 🎉

aronatkins commented:
Visual feedback for dark-mode. I was testing using Safari. None of this is must-fix; only recording my initial reactions. Also keep in mind that I am not the target audience for this theme.

### Code

The font/color choices for code blocks and inline code make them feel more prominent than the surrounding text. Here's one small example, but it also applies to other occurrences.

<img width="326" alt="image" src="https://github.com/posit-dev/product-doc-theme/assets/362187/31c99ee8-b776-4418-89c5-04f137fb33ec">

This felt especially pronounced with the "Output or filenames" code chunk; it drew my eye over everything else on the page.

### Table headers

Table headers feel more visually prominent in dark mode than in light mode. This may be because they use a brighter color than the surrounding text. The fact that they are bold makes them more of a visual draw than the accompanying section header.

<img width="697" alt="image" src="https://github.com/posit-dev/product-doc-theme/assets/362187/cc797444-eb11-4004-8d33-2ae710bce12b">

### Table row highlighting

Table row highlighting feels much more subtle in dark mode than in light mode.

<img width="691" alt="image" src="https://github.com/posit-dev/product-doc-theme/assets/362187/0dc9979a-bbd7-45b7-9c9a-63adfd14810b">

### Code-nested tabset

The nested tabset beneath the "code" pill is doing some strange things.

First, there appears to be a tiny underline lingering on the left, perhaps because the bubble around RHEL is too large?

<img width="173" alt="image" src="https://github.com/posit-dev/product-doc-theme/assets/362187/c3fff72d-5e39-4d58-aa08-d4fc520c2744">

Second, on mouseover, the whole set of headings shifts around. This screen recording shows the effect:

![tab-hop](https://github.com/posit-dev/product-doc-theme/assets/362187/33ed2f18-9149-4c37-bf47-e14ecf5123b2)

### Callout icons

The callout icons probably need to be a lighter color.

<img width="154" alt="image" src="https://github.com/posit-dev/product-doc-theme/assets/362187/4b7b802a-06dc-43fd-8735-6cfd87489e83">

aronatkins commented on changelog.md:
```diff
@@ -1,3 +1,14 @@
+
+Known issue:
+
+* Dark mode reloads cause flashing (Quarto has an open issue with plans to address in v1.5 or v16)
```
nits: Link to the tracking issue, and v1.6, not v16.

aronatkins commented on index-2.qmd:
```diff
@@ -0,0 +1,98 @@
+---
```
nit: Do you intend to link this page from `_quarto.yml` or from any other page? Also consider naming it something other than `index-2.qmd`.

AshleyHenry15 commented:
@aronatkins - The nesting of the pills and tabs gets weird.  I am assuming that it is due to the nesting of the different styles. I think that I will remove the example from the file for now until I can dive in and figure out if there is a way to make that work nicely.

- Table header color: I had changed it to white for accessibility but I just tested and it passes with the gray. So, I have removed the white coloring to the `th` class, and it should be fixed now.
- Table rows: Yeah, we went back and forth on the colors for this a lot. I used GitHub tables as an example but the blue complicates things. This will likely be something that I bring up with Veronica for some guidance at some point.
